torch/_jit_internal.py

```diff
@@ -658,7 +658,7 @@ def get_static_fn(cls, fn):
 
 
 def get_torchscript_modifier(fn):
-    if not callable(fn):
+    if not callable(fn) or isinstance(fn, torch.jit.RecursiveScriptClass):
```

Comment:
I am probably missing something, but why is this special case needed?

Based on my naive understanding, if we remove this special case, `get_torchscript_modifier` called on `torch.jit.RecursiveScriptClass` instances would return `FunctionModifiers.DEFAULT`. Then TorchScript would attempt to script the instance, which would hit line 1083 of `_script.py`. And the final result would be correct. Is this accurate or am I missing something obvious?

Reply:
yep deleted